plugins/jvm-example/src/main/java/org/elasticsearch/plugin/example/JvmExamplePlugin.java

```diff
@@ -111,5 +117,16 @@ public Settings additionalSettings() {
     public void onModule(RepositoriesModule repositoriesModule) {
     }
 
-
+    /**
+     * Module decalaring some example configuration and a _cat action that uses
+     * it.
+     */
+    public static class ConfiguredExampleModule extends AbstractModule {
+        @Override
+        protected void configure() {
+          bind(ExamplePluginConfiguration.class).asEagerSingleton();
+          Multibinder<AbstractCatAction> catActionMultibinder = Multibinder.newSetBinder(binder(), AbstractCatAction.class);
```

Comment:
+1 on registration directly via the `RestActionModule`

I don't think multiple multibindings is supported, but tbh I wouldn't do that if it is/was. By exposing the extension points as registration methods on the Module you communicate what can be extended and what not, what accepts multiple implementation of X and what just replaces a singleton of X, etc...